com.lunarclient.LunarClient.yml

```diff
@@ -0,0 +1,69 @@
+app-id: com.lunarclient.LunarClient
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+branch: stable
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+command: lunarclient
+separate-locales: false
+rename-icon: lunarclient
+finish-args:
+  - --persist=.minecraft
+  - --persist=.lunarclient
+  - --share=network
+  - --device=dri
+  - --socket=x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --persist=.java
+modules:
+  - name: unappimage
+    buildsystem: simple
+    build-commands:
+      - make -C squashfs-tools install INSTALL_DIR=/app/bin
+    sources:
+      - type: git
+        url: https://github.com/refi64/unappimage
+        commit: d7f86f2a0d7ec3a69211125207d5f127386b849a
+
+  - name: lunarclient
+    buildsystem: simple
+    cleanup:
+      - squashfs-root
+    sources:
+      - type: extra-data
+        url: https://launcherupdates.lunarclientcdn.com/Lunar%20Client-2.9.3.AppImage
```

Comment:
This should be configured to detect new versions with https://github.com/flathub/flatpak-external-data-checker/.

Reply:
Need YAML checker, impossible until https://github.com/flathub/flatpak-external-data-checker/issues/268

Reply:
This can be done using rotating-url which can work with app images.

https://github.com/flathub/flatpak-external-data-checker#url-checker

```
If the filename ends with .AppImage, the version number is extracted from the AppImage. (It is run in a bwrap sandbox.)
```

Reply:
> This can be done using rotating-url which can work with app images.
> 
> https://github.com/flathub/flatpak-external-data-checker#url-checker
> 
> ```
> If the filename ends with .AppImage, the version number is extracted from the AppImage. (It is run in a bwrap sandbox.)
> ```

Ah, may be it. Ignore my last comment as I didn’t see notification for this

Reply:
yeah it doesn't work
```
ERROR   src.checker: Error reading source: [ordereddict([('type', 'rotating-url'), ('url', 'https://launcherupdates.lunarclientcdn.com/Lunar%20Client-2.9.3.AppImage')])] is not of type 'object'

Failed validating 'type' in schema['properties']['x-checker-data']:
    {'properties': {'arches': {'items': {'type': 'string'},
                               'type': 'array'},
                    'is-main-source': {'type': 'boolean'},
                    'type': {'type': 'string'}},
     'required': ['type'],
     'type': 'object'}

On instance['x-checker-data']:
    [ordereddict([('type', 'rotating-url'), ('url', 'https://launcherupdates.lunarclientcdn.com/Lunar%20Client-2.9.3.AppImage')])]
INFO    src.checker: Checking 2 external data items
INFO    src.checker: Started check [1/2] git unappimage (from com.lunarclient.LunarClient.yml)
INFO    src.checkers.gitchecker: Skipping source unappimage, commit is specified, but neither tag nor branch
INFO    src.checker: Finished check [1/2] git unappimage (from com.lunarclient.LunarClient.yml)
INFO    src.checker: Skipped check [2/2] archive xrandr-1.5.1.tar.gz (from com.lunarclient.LunarClient.yml)
WARNING src.main: Check finished with 1 error(s)
```
I believe the appimage version is only used to construct a new version-specific url using the vendor's latest release url. Not to get the latest version from an old appimage somehow

Reply:
a new f-e-d-c PR was made with the required changes: [#270](https://github.com/flathub/flatpak-external-data-checker/pull/270)